todoapp/TODOapp/app/src/main/java/com/nearsoft/androidschool/todoapp/activities/detail/DetailActivity.java

```diff
@@ -5,50 +5,163 @@
-        dateTextView = (TextView) findViewById(R.id.date_text);
-        editFab = (FloatingActionButton) findViewById(R.id.fab);
+        titleEditTextView = (EditText) findViewById(R.id.title_field);
+        notesEditTextView = (EditText) findViewById(R.id.notes_field);
+        dateCardView = (CardView) findViewById(R.id.date_view);
+        editFab = (FloatingActionButton) findViewById(R.id.edit_fab);
+        saveFab = (FloatingActionButton) findViewById(R.id.save_fab);
+        dateTextView = (TextView) dateCardView.findViewById(R.id.date_text);
+        dateSwitch = (Switch) findViewById(R.id.switch_date);
+    }
+
+    private void setSaveListener() {
+        saveFab.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                try {
+                    saveData();
+                    Snackbar.make(editFab, "Data Saved (Not saving in reality)", Snackbar.LENGTH_SHORT).show();
+                    buttonViewConfig(false);
+                } catch (AssertionError er) {
+                    Snackbar.make(editFab, R.string.set_date_or_change_switch_message, Snackbar.LENGTH_SHORT).show();
+                }
+            }
+        });
+    }
+
+    private void setEditListener() {
         editFab.setOnClickListener(new View.OnClickListener() {
             @Override
+            public void onClick(View view) {
+                buttonViewConfig(true);
+            }
+        });
+    }
+
+    private void setDateListener() {
+        dateCardView.setOnClickListener(new View.OnClickListener() {
+            @Override
             public void onClick(View v) {
-                Snackbar.make(editFab, "should open the \"edit\" activity", Snackbar.LENGTH_SHORT).show();
+                showDatePicker();
             }
         });
+    }
 
-        titleTextView.setText(todo.getTitle());
-        notesTextView.setText(todo.getNotes());
-        dateTextView.setText(todo.getDate());
+    private void showDatePicker() {
+        DatePickerFragment dialog = new DatePickerFragment();
+        dialog.attachListener(new android.app.DatePickerDialog.OnDateSetListener() {
+            @Override
+            public void onDateSet(DatePicker datePicker, int year, int month, int dayOfMonth) {
+                Date dateTime = new Date(year, month + 1, dayOfMonth, 0, 0);
+                updateDate(dateTime);
+            }
+        });
+        dialog.show(this.getSupportFragmentManager(), "datePicker");
     }
 
-    private ToDoContent getTodo() {
+    private void setSwitchListener() {
+        dateSwitch.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
+                dateCardView.setVisibility(isChecked ? View.VISIBLE : View.INVISIBLE);
+            }
+        });
+    }
+
+    private void buttonViewConfig(boolean isEditClicking) {
```

Comment:
buttonViewConfig is a very bad name for a method, something like "enableEditMode" works better for this.